### PR TITLE
refactor: generate uuid in service

### PR DESCRIPTION
**Pull request by BaojunCZ**

### CODE REVIEW

File: packages/neuron-wallet/tests/store/wallet_store.js [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-    master: {
-      privateKey: 'privateKey2',
-      chainCode: 'chainCode2',
+  const wallet2 = {
+    id: uuid.v4(),
+    name: 'wallet2',
+    keystore: {
+      version: 0,
+      id: '1',
+      crypto: { cipher: 'wallet1', cipherparams: null, ciphertext: 'wallet2', kdf: '2', kdfparams: null, mac: '2' },
```
Comment (Keith-CY) [inline packages/neuron-wallet/tests/store/wallet_store.js]: `cipher: 'wallet1'` or `cipher: 'wallet2'`

File: packages/neuron-wallet/tests/store/wallet_store.js [hunk 2]
```diff
@@ -33,37 +38,25 @@ describe('wallet store', () => {
...
-      keystore: mKeystore1,
-    })
+    assert.deepEqual(wallet1, wallet1Store)
+    assert.deepEqual(wallet2, wallet2Store)
   })
 
   it('get not exist wallet', () => {
-    walletStore.saveWallet(mWalletName1, mKeystore1)
-    walletStore.saveWallet(mWalletName1, mKeystore1)
+    walletStore.saveWallet(wallet1)
```
Comment (Keith-CY) [inline packages/neuron-wallet/tests/store/wallet_store.js]: why was the wallet1 saved twice?

Comment (BaojunCZ) [reply packages/neuron-wallet/tests/store/wallet_store.js]: I have removed.

File: packages/neuron-wallet/src/store/WalletStore.ts [hunk 3]
```diff
@@ -51,18 +50,12 @@ export default class WalletStore {
...
-    const walletId = v4()
+  saveWallet = (walletData: WalletData): string => {
     let idList = this.getIDList()
-    const walletData = {
-      id: walletId,
-      name: walletName,
-      keystore: walletKeystore,
-    }
-    idList = idList.concat(walletId)
+    idList = idList.concat(walletData.id)
```
Comment (ashchan) [inline packages/neuron-wallet/src/store/WalletStore.ts]: Better extract id list update into a separate method.

Comment (BaojunCZ) [reply packages/neuron-wallet/src/store/WalletStore.ts]: ok

File: packages/neuron-wallet/src/store/WalletStore.ts [hunk 4]
```diff
@@ -38,8 +38,14 @@ export default class WalletStore {
     return this.walletIDStore.get(WalletIDKey, [])
   }
 
-  private setIDList = (list: string[]) => {
-    this.walletIDStore.set(WalletIDKey, list)
+  private addIDList = (id: string) => {
```
Comment (ashchan) [inline packages/neuron-wallet/src/store/WalletStore.ts]: Perhaps `addWalletID` and `removeWalletID`? It's not adding/deleting ID list, but ID (single entry).

Comment (BaojunCZ) [reply packages/neuron-wallet/src/store/WalletStore.ts]: ok

File: packages/neuron-wallet/src/services/wallets.ts [hunk 5]
```diff
@@ -55,7 +56,7 @@ export default class WalletService {
   }
 
   public create = ({ name, keystore }: { name: string; keystore: Keystore }): Wallet => {
-    const id = walletStore.saveWallet(name, keystore)
+    const id = walletStore.saveWallet({ id: v4(), name, keystore })
```
Comment (ashchan) [inline packages/neuron-wallet/src/services/wallets.ts]: Now that we pass in pre-generated ID, it should always return a valid ID so no need to check `if (id)`, right?

`saveWallet` doesn't have to return the `ID` anymore.

Comment (BaojunCZ) [reply packages/neuron-wallet/src/services/wallets.ts]: Right

**Comment by Keith-CY** [comment]
CI failed

File: packages/neuron-wallet/src/services/wallets.ts [hunk 6]
```diff
@@ -56,15 +56,13 @@ export default class WalletService {
...
-    const id = walletStore.saveWallet({ id: v4(), name, keystore })
-    if (id) {
-      const storedWallet = walletStore.getWallet(id)
-      return {
-        id,
-        name: storedWallet.name,
-      }
+    const id = v4()
+    walletStore.saveWallet({ id, name, keystore })
+    const storedWallet = walletStore.getWallet(id)
```
Comment (Keith-CY) [inline packages/neuron-wallet/src/services/wallets.ts]: The `storedWallet` is equal to `{id, name, keystore}`. Refactor it later.

Comment (BaojunCZ) [reply packages/neuron-wallet/src/services/wallets.ts]: ok

**Pull request merged**